### DISCUSSION TITLE
How ger collection name

### DISCUSSION THREAD

**Discussion by ViktorKirichenko** [xunit/xunit#2504] [Question] [answered]
hello, How i can get collection name when test is run?

```
[Collection("Example Collection Name")]
public class testClassname: FixtureBase, IDisposable {

    [Fact]
    public void Test1() {
		<Get Collection name>
	}
```

**Comment by bradwilson** [comment, marked as answer]
There is no way to do this today in v2.

In v3, you can get this from [TestContext](https://github.com/xunit/xunit/blob/c27a91f8cbcee37cb45699a3a81287bca225e876/src/xunit.v3.core/TestContext.cs).[Current](https://github.com/xunit/xunit/blob/c27a91f8cbcee37cb45699a3a81287bca225e876/src/xunit.v3.core/TestContext.cs#L45).[TestCollection](https://github.com/xunit/xunit/blob/c27a91f8cbcee37cb45699a3a81287bca225e876/src/xunit.v3.core/TestContext.cs#L112).[DisplayName](https://github.com/xunit/xunit/blob/c27a91f8cbcee37cb45699a3a81287bca225e876/src/xunit.v3.core/Sdk/v3/TestCases/_ITestCollection.cs#L19) (or by injecting [ITestContextAccessor](https://github.com/xunit/xunit/blob/c27a91f8cbcee37cb45699a3a81287bca225e876/src/xunit.v3.core/ITestContextAccessor.cs) in your test class's constructor, if you would prefer a more DI-friendly way: [example](https://github.com/xunit/xunit/blob/c27a91f8cbcee37cb45699a3a81287bca225e876/src/xunit.v3.core.tests/Acceptance/Xunit3AcceptanceTests.cs#L666-L682))
